Modules/OpenGhostDir.cs

```diff
@@ -10,50 +10,146 @@ internal class OpenGhostDir : Module
+                    ghostButton = ghostObject.GetComponent<MenuButtonHolder>();
+                    var backComponent = ghostObject.GetComponent<BackButtonAccessor>();
+                    if (backComponent)
+                        UnityEngine.Object.Destroy(backComponent);
+
+                    ghostButton.ButtonRef.onClick.RemoveAllListeners();
+                    ghostButton.ButtonRef.onClick.AddListener(() => Process.Start("file://" + GetGhostDirectory(level)));
+                }
+
+                ghostButton.name = "Button Ghost";
+                ghostButton.buttonText = "Open Ghost Directory";
+                ghostButton.buttonTextRef.text = "Open Ghost Directory";
+                ghostButton.gameObject.SetActive(_setting_GhostButton.Value);
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError("error on setting up the button " + e);
+            }
+        }
+
         [HarmonyPrefix]
         [HarmonyPatch(typeof(MenuScreenResults), "OnSetVisible")]
         private static void OnMenuScreen(ref MenuScreenResults __instance)
         {
-            if (ghostButton == null)
+            try
             {
+                if (ghostButton && __instance.buttonsToLoad.Contains(ghostButton))
+                    __instance.buttonsToLoad.Remove(ghostButton);
                 var button = __instance._buttonContine;
-                // copy the layout
-                var layout = UnityEngine.Object.Instantiate(button.transform.parent.gameObject, button.transform.parent.parent);
-                layout.name = "Ghost Button Holder";
-                // empty it
-                foreach (Transform child in layout.transform)
-                    UnityEngine.Object.Destroy(child.gameObject);
-
-                // copy the button and put it in the new layout
-                ghostButton = UnityEngine.Object.Instantiate(button.gameObject, layout.transform).GetComponent<MenuButtonHolder>();
-                ghostButton.name = "Button Ghost";
-                ghostButton.buttonText = "Open Ghost Directory";
-                ghostButton.buttonTextRef.text = "Open Ghost Directory";
+                SetupButton(button);
+
                 var pos = button.transform.parent.position;
                 pos.x = -0.35f; // don't ask me how. the math just ISN'T THERE i had to hardcode it
-                layout.transform.position = pos;
+                ghostButton.transform.parent.position = pos;
 
-                ghostButton.ButtonRef.onClick.RemoveAllListeners();
-                ghostButton.ButtonRef.onClick.AddListener(() => Process.Start("file://" + GetGhostDirectory()));
+                if (!__instance.buttonsToLoad.Contains(ghostButton))
+                    __instance.buttonsToLoad.Add(ghostButton);
             }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogError("error on OnMenuScreen " + e);
```

Comment:
For debug reasons I recommend e.printStackTrace(). It's way better

Reply:
See above.